OrbitSsh/SocketTests.cpp

```diff
@@ -218,7 +218,12 @@ TEST(Socket, Shutdown) {
   // send shutdown client
   ASSERT_TRUE(client_socket.value().Shutdown());
 
-  // server should have an immidiate result with WaitDisconnect
+  // Even though this is only a local connection, it might take a split second.
+  if (OrbitSsh::shouldITryAgain(client_socket.value().CanBeRead())) {
```

Comment:
Could/should this be a while loop to make sure you end up in the desired state? (If this never happens, the test will timeout, which I think is the desired result.)

Reply:
Yes, a timeout is probably a better idea. Going to implement it

Reply:
Tests should already have a timeout, though I'm not sure how to set this (normally this is done in build files).

Reply:
done. Ctest should usually have a default timeout of 30 seconds, but I tried it and it did not work. I added a 10 seconds timeout to the cmake file and tested that it works. Also changed the test cases

Reply:
AFAIK ctest has a default timeout of 1500secs.

Reply:
You are correct. My mistake